weblate_schemas/messages.py

```diff
@@ -0,0 +1,126 @@
+# Copyright © Michal Čihař <michal@weblate.org>
+#
+# SPDX-License-Identifier: MIT
+
+"""Message class implementation for Weblate Fedora Messaging."""
+
+from __future__ import annotations
+from datetime import datetime
+
+from . import load_schema
+from fedora_messaging import message
+
+
+class BaseMessage(message.Message):
+    """Inherit the Message class from fedora_messaging."""
+
+    def __init__(self, **kwargs) -> None:
+        """Initialize BaseMessage class."""
+        super().__init__(**kwargs)
+
+    @property
+    def app_name(self) -> str:
+        """Return the app name."""
+        return "Weblate"
+
+    @property
+    def app_icon(self) -> str:
+        """Return the App icon URL."""
+        return "https://weblate.org/static/weblate-128.png"
+
+
+class WeblateV1Message(BaseMessage):
+    """Actual Weblate message class which uses the Messaging schema."""
+
+    def __init__(self, **kwargs) -> None:
+        """Initialize the WeblateMessage class with loading of the body_schema."""
+        super().__init__(**kwargs)
+        self.body_schema = load_schema("weblate-messaging.schema.json")
+
+    @property
+    def agent_name(self) -> str:
+        """The username who cause the action."""
+        return self.body.get("user", "")
+
+    @property
+    def change_id(self) -> int:
+        """Return the change ID."""
+        return self.body.get("change_id")
```

Comment:
According to the JSON schema, `change_id` is required, you can be sure it'll always be there, so you can use `self.body["change_id"]`. (just a suggestion, `get()` will work too)